frontend/class-breadcrumbs.php

```diff
@@ -145,6 +144,28 @@ public function __toString() {
 		return self::$before . $this->output . self::$after;
 	}
 
+	/**
+	 * Returns the link url for a single id.
+	 *
+	 * When the target is private and the user isn't allowed to access it, just return an empty string.
+	 *
+	 * @param int $id The target id.
+	 *
+	 * @return string Empty string when post isn't accessible. An URL if accessible.
+	 */
+	protected function get_link_url_for_id( $id ) {
+		$status = get_post_status( $id );
+		$cpt 	= get_post_type( $id );
+		$url    = get_permalink( $id );
```

Comment:
As `get_permalink` is a fairly heavy call, this should be only called whenever the other conditions are pointing towards using the link for the referenced object.

Doing a `private` check before doing the URL check would improve this functionality.